### DISCUSSION TITLE
Query Template property required to show Private & Shared Content in OneDrive?

### DISCUSSION THREAD

**Discussion by Ujjwal06it** [microsoft-search/pnp-modern-search#3706] [General]
Hi Guys,
Do you know what should be the Query Template property for {searchTerms} in order to show "**Shared by me/Shared with me**" in OneDrive/SPO?

I have used -sharedwithusersowsuser:{User.Name} which does not seems to be providing expected results. Those content which has been shared with me through "Groups" or "Links" or "shared through Teams Chat to me" are also appearing in the search results. Is there any way I can fix this? and what is the way to findout the content which I have shared to others?

Your help is highly appreciated. @wobba especially looking for your expertise ;-)

**Comment by wobba** [comment]
Sharing a pro tip :) `PrivacyIndicator:0` will show files only the person searching has access to. Content you have shared or is shared with you is not doable 100% via search API's - as they work on access control.

For sharing you need to look into the Graph insights API https://learn.microsoft.com/en-us/graph/api/insights-list-shared?view=graph-rest-1.0&tabs=http

**Comment by Ujjwal06it** [comment]
Thank you @wobba but I am using SharePoint Search and using the KQL. wondering how I can see only my files, not shared with me by the other users.

**Comment by wobba** [comment]
How do you define "your files"? It has a semantic meaning which you have to expand on. Then I'll try to translate it to KQL. And not every thing is solved by search API's.
